spec/controllers/users_controller_spec.rb

```diff
@@ -25,56 +25,103 @@
-      expect(joe.follows?(alice)).to be_truthy
-    end
+      it "saves @user.id to session[:user_id] if @user is saved" do
+        #post :create, user: {username: "Bob Builder", email: "bob_b@bobb.com", password: "secret"}
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(session[:user_id]).to eq(User.first.id)
+      end
+
+      it "makes the user follow the inviter" do
+        alice = Fabricate(:user)
+        invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
+        post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
+        joe = User.where(email: 'joe@example.com').first
+        expect(joe.follows?(alice)).to be_truthy
+      end
+
+      it "makes the inviter follow the user" do
+        alice = Fabricate(:user)
+        invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
+        post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
+        joe = User.where(email: 'joe@example.com').first
+        expect(alice.follows?(joe)).to be_truthy
+      end
+      
+      it "expires the invitation upon acceptance" do
+        alice = Fabricate(:user)
+        invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
+        post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
+        expect(Invitation.first.token).to be_nil
+      end
+
+      it "redirects to home_path if @user is saved" do 
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(response).to redirect_to home_path
+      end
 
-    it "makes the inviter follow the user" do
-      alice = Fabricate(:user)
-      invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
-      post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
-      joe = User.where(email: 'joe@example.com').first
-      expect(alice.follows?(joe)).to be_truthy
     end
     
-    it "expires the invitation upon acceptance" do
-      alice = Fabricate(:user)
-      invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
-      post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
-      expect(Invitation.first.token).to be_nil
+    context "valid personal info and declined card" do
+      it "does not create a new user record" do
+        charge = double(:charge, successful?: false, error_message: "Your card was declined.")
+        StripeWrapper::Charge.should_receive(:create).and_return(charge)
+        post :create, user: Fabricate.attributes_for(:user), stripeToken: '1231241'
```

Comment:
I would consider using `let` `before`/`do` etc to simplify this. You really are just doing the same thing every time so it makes sense to not define it multiple times.